zkevm-circuits/src/blobkzg_circuit/src/main.rs

```diff
@@ -0,0 +1,585 @@
+                                integer_chip.sub(&mut ctx, &assigned_xns[256], &assigned_one)?;
+                            ctx.next();
+                            integer_chip.div(&mut ctx, &numerator, &assigned_n)?
+                        };
+                        ctx.next();
+                        integer_chip.mul(&mut ctx, &assigned_evals[4096], &multiplier)?
+                    };
+                    ctx.next();
+                    integer_chip.assert_equal(&mut ctx, &assigned_evals[4097], &eval_last)?;
+                }
+
+                // 3. ADD CONSTRAINTS FOR EACH ROOT OF UNITY STEP
+                for i in 0..4095 {
+                    let wi_next = integer_chip.mul(&mut ctx, &assigned_wis[i], &assigned_w)?;
+                    ctx.next();
+                    integer_chip.assert_equal(&mut ctx, &assigned_wis[i + 1], &wi_next)?;
+                }
+
+                // 4. ADD CONSTRAINTS FOR INTERMEDIATE EXPONENTIATION STEPS
+                for i in 0..256 {
+                    let xn_next = {
+                        let multiplier = {
+                            let important_bit = integer_chip.mul(
+                                &mut ctx,
+                                &assigned_x2is[i],
+                                &assigned_nbits[i],
+                            )?;
+                            ctx.next();
+                            let not_important_bit =
+                                integer_chip.sub(&mut ctx, &assigned_one, &assigned_nbits[i])?;
+                            ctx.next();
+                            integer_chip.add(&mut ctx, &important_bit, &not_important_bit)?
+                        };
+                        ctx.next();
+                        integer_chip.mul(&mut ctx, &assigned_xns[i], &multiplier)?
+                    };
+                    ctx.next();
+                    integer_chip.assert_equal(&mut ctx, &assigned_xns[i + 1], &xn_next)?;
+                }
+
+                // 5. ADD CONSTRAINTS FOR EACH x^(2^i) STEP
+                for i in 0..255 {
+                    let x2i_next =
+                        integer_chip.mul(&mut ctx, &assigned_x2is[i], &assigned_x2is[i])?;
+                    ctx.next();
+                    integer_chip.assert_equal(&mut ctx, &assigned_x2is[i + 1], &x2i_next)?;
+                }
+                Ok(())
+            },
+        )?;
+        Ok(())
+    }
+}
+
+/// Native Field (likely BN256)
+type N = halo2_proofs::halo2curves::bn256::Fr;
+
+/// Wrong Field (BLS12-381)
+type W = halo2_proofs::halo2curves::bls12_381::Scalar;
+
+/// ROOT OF UNITY WITH ORDER 4096: 0x564c_0a11_a0f7_04f4_fc3e_8acf_e0f8_245f_0ad1_347b_378f_bf96_e206_da11_a5d3_6306
+const ROOT_OF_UNITY: W = W::from_raw([
+    0xe206_da11_a5d3_6306,
+    0x0ad1_347b_378f_bf96,
+    0xfc3e_8acf_e0f8_245f,
+    0x564c_0a11_a0f7_04f4,
+]);
+const ROOT_OF_UNITY_ORDER: W = W::from_raw([0x1000, 0x0, 0x0, 0x0]);
+
+const NUMBER_OF_POINTS: usize = 4096;
+
+fn main() {
```

Comment:
It's unclear why we have a main function testing hardcoded values. Maybe move this to a test case?